conda/plugins/manager.py

```diff
@@ -159,6 +161,24 @@ def get_solver_backend(self, name: str = None) -> type[Solver]:
 
         return backend
 
+    def run_pre_command_hooks(self, command: str, args: Namespace) -> None:
```

Comment:
I think this function would be better suited for the subject code where commands are called, than the plugin manager since the plugin manager exists to query plugin metadata, not handle actions of hook implementations. Or maybe you could split it between a function for getting a list of `pre_commands` and a function to call them in the CLI code?

Reply:
Main reason: separation of concerns between plugin code and CLI code for example.